ksql-common/src/main/java/io/confluent/ksql/util/DecimalUtil.java

```diff
@@ -153,6 +153,28 @@ public static BigDecimal ensureFit(final BigDecimal value, final Schema schema)
     }
   }
 
+  /**
+   * Converts a schema to a decimal schema with set precision/scale without losing
+   * scale or precision.
+   *
+   * @param schema the schema
+   * @return the decimal schema
+   * @throws KsqlException if the schema cannot safely be converted to decimal
+   */
+  public static Schema toDecimal(final Schema schema) {
+    switch (schema.type()) {
+      case BYTES:
+        requireDecimal(schema);
+        return schema;
+      case INT32:
+        return builder(10, 0).build();
+      case INT64:
+        return builder(19, 0).build();
+      default:
```

Comment:
What about BOOLEAN? is it needed?
`return builder(1, 0).build()`

And DOUBLE? It has a fixed 15 precision and 15 scale

Reply:
Since we don't support boolean math anywhere I didn't support that. For DOUBLE, I did some research and that is not what Postgres does, math between decimal and float64 converts everything to float64 - in fact, DECIMAL(15, 15) makes no sense from my point of view since that means I could only represent things less than 1.